### PR TITLE
Rebuild layer if npm config/node version changes

### PR DESCRIPTION
**Pull request by andymoe**
fixes gh-153

### CODE REVIEW

File: build_process_resolver.go [hunk 1]
```diff
@@ -101,6 +103,36 @@ func (r BuildProcessResolver) Resolve(workingDir, cacheDir string) (BuildProcess
...
 
+// cacheExecutableResponse writes the output of a successfully executed command
+// to a tmp file and returns the file location and possibly and error
+func cacheExecutableResponse(executable Executable, args []string, workingDir string, logger scribe.Logger) (string, error) {
+	buffer := bytes.NewBuffer(nil)
+	err := executable.Execute(pexec.Execution{
+		Args:   args,
+		Dir:    workingDir,
+		Stdout: buffer,
+		Stderr: buffer,
```
Comment (ryanmoran) [inline build_process_resolver.go:114]: Do we want to store both `stdout` and `stdin`? It looks like `stderr` might include output that changes even when the environment configuration isn't changing. For example, I get the following when I run `npm config list` on a simple app:

```
; "builtin" config from /usr/local/lib/node_modules/npm/npmrc

prefix = "/usr/local"

; node bin location = /usr/local/Cellar/node/16.1.0/bin/node
; cwd = /Users/rmoran/workspace/paketo-buildpacks/npm-install/integration/testdata/simple_app
; HOME = /Users/rmoran
; Run `npm config ls -l` to show all defaults.
npm notice
npm notice New minor version of npm available! 7.11.2 -> 7.13.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.13.0
npm notice Run npm install -g npm@7.13.0 to update!
npm notice
```

The lines prefixed with `npm notice` are all printed to `stderr` and they contain text that may change in cases we shouldn't expect to do a rebuild. For example, if NPM releases `v7.14.0`, then the output would change to
```
npm notice
npm notice New minor version of npm available! 7.11.2 -> 7.14.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.14.0
npm notice Run npm install -g npm@7.14.0 to update!
npm notice
```
None of that should have any impact on a rebuild. Its just a notice to tell you that a newer version is available.

**Comment by ryanmoran** [comment]
I noticed that `npm config list` may take an optional `--json` flag that seems to output more structured data. Any reason we might want to record the unstructured form over what we get from the `--json` output?

File: build_process_resolver.go (hunk 1 above)
Comment (andymoe) [reply build_process_resolver.go:114]: In addition to `npm config list --json` Some other options are:
```
‣ npm get node-version
14.16.1
```
or 

```
 ‣ npm get user-agent
npm/6.14.12 node/v14.16.1 linux x64
```

Originally I had `npm get node-version` however we changed it to be more consistent with the yarn-install build pack which appears to be problematic as you pointed out.

Personally, I'm happy to use any of the three above options - let me know if you or anyone else who sees this has a strong preference for one of them.

**Review by thitch97 — APPROVED** [review]

**Comment by thitch97** [comment]
Looks like the change to `npm get user-agent` may alleviate your concerns @ryanmoran. If so, I can go ahead and merge this.

**Pull request merged**